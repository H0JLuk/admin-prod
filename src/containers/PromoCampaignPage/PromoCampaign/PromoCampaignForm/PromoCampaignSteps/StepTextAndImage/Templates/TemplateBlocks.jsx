import React from 'react';
import cn from 'classnames';
import styles from './Template.module.css';

export const TextBlock = ({ label, text }) => (
    <>
        <div className={ styles.title }>{ label }</div>
        <div>{ text }</div>
    </>
);

export const ImageBlock = ({ label, type, src, size }) => (
    <>
        <div className={ styles.title }>
            { label }
        </div>
        <div className={ styles.block }>
            <div
                className={ cn({
                    [styles.logo]: type === 'logo',
                }) }
            >
                <img
                    className={ cn({
                        [styles.banner]: type === 'banner',
                        [styles.imageLogo]: type === 'logo',
                    }) }
                    src={ src }
                    alt=""
                />
            </div>
            <div className={ styles.footer }>
                <div className={ styles.fileName }>
                    { (src || '').split('/').pop() }
                </div>
                { /* FIXME: Добавить размер файла, когда определимся откуда мы его будем брать (с бэка или из Blob?) */ }
                <div className={ styles.fileSize }>{ size }</div>
            </div>
        </div>
    </>
);