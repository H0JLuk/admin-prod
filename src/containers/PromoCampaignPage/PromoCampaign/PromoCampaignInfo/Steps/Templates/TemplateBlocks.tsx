import React from 'react';
import cn from 'classnames';

import styles from './Template.module.css';

interface ITextBlock {
    label: string;
    text: string;
}

type IImageBlockProps = {
    label?: string;
    type?: 'logo' | 'banner' | 'text';
    src?: string;
    size?: string | number;
};

export const TextBlock: React.FC<ITextBlock> = ({ label, text }) => (
    <>
        <div className={styles.title}>{label}</div>
        <div className={styles.text}>{text}</div>
    </>
);

export const ImageBlock: React.FC<IImageBlockProps> = ({ label, type, src, size }) => (
    <>
        <div className={styles.title}>
            {label}
        </div>
        <div className={styles.block}>
            <div
                className={cn({
                    [styles.logo]: type === 'logo',
                })}
            >
                <img
                    className={cn({
                        [styles.banner]: type === 'banner',
                        [styles.imageLogo]: type === 'logo',
                    })}
                    src={src}
                    alt=""
                />
            </div>
            <div className={styles.footer}>
                <div className={styles.fileName}>
                    {(src || '').split('/').pop()}
                </div>
                <div className={styles.fileSize}>{size}</div>
            </div>
        </div>
    </>
);
