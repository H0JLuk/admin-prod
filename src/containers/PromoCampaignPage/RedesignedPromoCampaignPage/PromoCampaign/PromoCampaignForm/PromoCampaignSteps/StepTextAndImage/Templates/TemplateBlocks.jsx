import React from 'react';
import cn from 'classnames';

import styles from './Template.module.css';

export const TextBlock = ({ label, text }) => (
    <>
        <div className={ styles.title }>{ label }</div>
        <div>{ text }</div>
    </>
);

export const ImageBlock = ({ label, type, src, size }) => (
    <>
        <div className={ styles.title }>
            { label }
        </div>
        <div className={ styles.block }>
            <div
                className={ cn({
                    [styles.icon]: type === 'icon',
                }) }
            >
                <img
                    className={ cn({
                        [styles.image]: type === 'image',
                        [styles.imageIcon]: type === 'icon',
                    }) }
                    src={ src }
                    alt=""
                />
            </div>
            <div className={ styles.footer }>
                <div className={ styles.fileName }>
                    { (src || '').split('/').pop() }
                </div>
                {/* TODO: Добавить размер файла, когда определимся откуда мы его будем брать (с бэка или из Blob?) */}
                <div className={ styles.fileSize }>{ size }</div>
            </div>
        </div>
    </>
);