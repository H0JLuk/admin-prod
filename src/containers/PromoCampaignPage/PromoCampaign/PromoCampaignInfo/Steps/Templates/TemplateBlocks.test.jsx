import React from 'react';
import { ImageBlock, TextBlock } from './TemplateBlocks';
import { shallow } from 'enzyme';

describe('<TemplateBlocks /> test', () => {
    describe('<ImageBlock /> test', () => {
        const props = {
            label: 'label',
            type: 'banner',
            src: 'src',
            size: 'size',
        };
        const ImageBlockItem = shallow(<ImageBlock { ...props } />);

        it('ImageBlock snapshot', () => {
            expect(ImageBlockItem.html()).toMatchSnapshot();
        });

        it('check banner type', () => {
            expect(ImageBlockItem.find('.banner').hasClass('banner')).toEqual(true);
        });

        it('check logo type', () => {
            const newProps = {
                ...props,
                type: 'logo',
            };
            ImageBlockItem.setProps(newProps);
            expect(ImageBlockItem.find('.logo').hasClass('logo')).toEqual(true);
            expect(ImageBlockItem.find('.imageLogo').hasClass('imageLogo')).toEqual(true);
        });
    });

    describe('<TextBlock /> test', () => {
        const props = {
            label: 'label',
            text: 'text',
        };
        const TextBlockItem = shallow(<TextBlock { ...props } />);

        it('TextBlock snapshot', () => {
            expect(TextBlockItem.html()).toMatchSnapshot();
        });

        it('should render text and label', () => {
            expect(TextBlockItem.find('.title').text()).toBe(props.label);
            expect(TextBlockItem.find('.text').text()).toBe(props.text);
        });
    });
});
