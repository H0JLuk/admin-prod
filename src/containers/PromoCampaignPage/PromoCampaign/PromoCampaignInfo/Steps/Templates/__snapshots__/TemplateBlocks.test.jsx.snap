// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateBlocks /> test <ImageBlock /> test ImageBlock snapshot 1`] = `"<div class=\\"title\\">label</div><div class=\\"block\\"><div class=\\"\\"><img class=\\"banner\\" src=\\"src\\" alt=\\"\\"/></div><div class=\\"footer\\"><div class=\\"fileName\\">src</div><div class=\\"fileSize\\">size</div></div></div>"`;

exports[`<TemplateBlocks /> test <TextBlock /> test TextBlock snapshot 1`] = `"<div class=\\"title\\">label</div><div class=\\"text\\">text</div>"`;
